lumigator/python/mzai/backend/backend/services/workflows.py

```diff
@@ -0,0 +1,166 @@
+import asyncio
+import datetime
+import uuid
+from collections.abc import Callable
+from uuid import UUID
+
+import loguru
+from fastapi import BackgroundTasks
+from lumigator_schemas.jobs import (
+    JobEvalLiteCreate,
+    JobInferenceCreate,
+    JobStatus,
+)
+from lumigator_schemas.workflows import WorkflowCreate, WorkflowResponse
+
+from backend.services.datasets import DatasetService
+from backend.services.jobs import JobService
+
+
+class WorkflowService:
+    def __init__(
+        self,
+        job_service: JobService,
+        dataset_service: DatasetService,
+    ):
+        self._job_service = job_service
+        self._dataset_service = dataset_service
+
+    NON_TERMINAL_STATUS = [
+        JobStatus.CREATED.value,
+        JobStatus.PENDING.value,
+        JobStatus.RUNNING.value,
+    ]
+    """list: A list of non-terminal job statuses."""
```

Comment:
don't think we need this comment, or let's move above the variable for consistency

Reply:
This is a copy past from some previous work, not anything new I added

Reply:
It can be updated to make it clearer for sure. But, I'd leave it in and where it is, as it needs to be below the class attribute so that IDEs etc. pick up the docs to show in IntelliSense (not sure what the generic term is).

![image](https://github.com/user-attachments/assets/754df526-369b-425f-bdfe-54aa6be51473)